docs/api.rst

```diff
@@ -2,3 +2,4 @@ Reference/API
 =============
 
 .. automodapi:: hips
+.. automodapi:: hips.tiles
```

Comment:
I would suggest that, at least for now, we import everything (except the utils) into the top-level `hips` namespace (like Numpy does it) instead of splitting things in sub-packages (like Scipy does it). The sub-packages and sub-files are only to organise the code into multiple files, but users then import everything from the `hips` namespace.
So instead of adding the automodapi directive here, I'd suggest you add a line `from .tiles import *` here:
https://github.com/hipspy/hips/blob/master/hips/__init__.py#L21
Also, there is now a merge conflict for `docs/api.rst`. Please rebase this PR against latest master.